core/src/main/java/com/github/shadowsocks/bg/ProxyInstance.kt

```diff
@@ -114,27 +114,25 @@ class ProxyInstance(val profile: Profile, private val route: String = profile.ro
 
         this.configFile = configFile
         val config = profile.toJson()
-        plugin?.let { (path, opts) -> config.put("plugin", path).put("plugin_opts", opts.toString()) }
+        val vpnFlags = if (service.isVpnService()) ";V" else ""
+        plugin?.let { (path, opts) -> config.put("plugin", path).put("plugin_opts", opts.toString() + vpnFlags) }
+        config.put("local_address", DataStore.listenAddress)
+        config.put("local_port", DataStore.portProxy)
         configFile.writeText(config.toString())
 
-        val cmd = service.buildAdditionalArguments(arrayListOf(
+        val cmd = arrayListOf(
                 File((service as Context).applicationInfo.nativeLibraryDir, Executable.SS_LOCAL).absolutePath,
-                "-b", DataStore.listenAddress,
-                "-l", DataStore.portProxy.toString(),
-                "-t", "600",
-                "-S", stat.absolutePath,
-                "-c", configFile.absolutePath))
+                "--stat-path", stat.absolutePath,
+                "-c", configFile.absolutePath)
+        if (service.isVpnService()) cmd += arrayListOf("--protect-path", "protect_path")
```

Comment:
I think we hardcoded the paths `./protect_path` and `./stats_path` in shadowsocks-libev. I don't think it would be useful to have an option for customization here...

Reply:
Okay, let's use hardcoded path in shadowsocks-rust as well.